src/main/kotlin/fi/metatavu/metaform/server/rest/TemplatesApi.kt

```diff
@@ -0,0 +1,139 @@
+package fi.metatavu.metaform.server.rest
+
+import fi.metatavu.metaform.api.spec.model.*
+import fi.metatavu.metaform.server.controllers.*
+import fi.metatavu.metaform.server.exceptions.DeserializationFailedException
+import fi.metatavu.metaform.server.exceptions.MalformedMetaformJsonException
+import fi.metatavu.metaform.server.rest.translate.TemplateTranslator
+import org.slf4j.Logger
+import java.util.*
+import javax.enterprise.context.RequestScoped
+import javax.inject.Inject
+import javax.transaction.Transactional
+import javax.ws.rs.core.Response
+
+@RequestScoped
+@Transactional
+@Suppress("unused")
+class TemplatesApi: fi.metatavu.metaform.api.spec.TemplatesApi, AbstractApi() {
+
+  @Inject
+  lateinit var logger: Logger
+
+  @Inject
+  lateinit var exportThemeController: ExportThemeController
+
+  @Inject
+  lateinit var templateController: TemplateController
+
+  @Inject
+  lateinit var templateTranslator: TemplateTranslator
+
+  override fun createTemplate(template: Template): Response {
+    val userId = loggedUserId ?: return createForbidden(UNAUTHORIZED)
+
+    if (!isMetatavuAdmin && !isRealmSystemAdmin) {
+      return createForbidden(createNotAllowedMessage(CREATE, TEMPLATE))
+    }
+
+    val templateData = template.data
```

Comment:
Data should be marked as required in the spec instead of checking it here manually